arm-containerregistry/2017-03-01/swagger/containerregistry.json

```diff
@@ -0,0 +1,746 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2017-03-01",
+    "title": "ContainerRegistryManagementClient"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.ContainerRegistry/checkNameAvailability": {
+      "post": {
+        "tags": [
+          "Operation"
+        ],
+        "description": "Checks whether the container registry name is available for use. The name must contain only alphanumeric characters, be globally unique, and between 5 and 60 characters in length.",
```

Comment:
You can put pattern, min/max length, like in https://github.com/Azure/azure-rest-api-specs/blob/886231c41eee52d2cd3a05127875429f51d26c16/arm-servermanagement/2015-07-01-preview/swagger/servermanagement.json#L132

This will do client side check and reduce traffic to your website in case names are not good, you can CRUD operations as well. Better to define a parameter for registryName in the end of file.